omod/src/main/webapp/resources/scripts/taggedObjects.js

```diff
@@ -0,0 +1,42 @@
+
+var app = angular.module('taggedObjects',['personService','tagService']);
+
+app.config(['$locationProvider', function($locationProvider){
+    $locationProvider.html5Mode({
+        enabled: true,
+        requireBase: false
+    });
+}]);
+
+app.controller('taggedObjectsCtrl',['$scope','$location','$q','Person','TagService',
+    function ($scope,$location,$q,Person,TagService) {
+
+    var tags = [];
+    $scope.tag = $location.search().tag;
+    var loadTags =function () {
+        var payload = {
+            objectType: 'org.openmrs.Patient',
+            tag: $scope.tag,
+            v: 'full'
+        };
+        TagService.getTags(payload).then(function (res) {
+            tags = res;
+            loadPatients();
+        });
+    };
+
+    $scope.patients = [];
+    var loadPatients = function () {
+        var promiseList = [];
+        angular.forEach(tags, function (t) {
+            var payload = {
+                uuid: t.objectUuid,
+            }
+           promiseList.push(Person.get(payload));
+        });
+        $q.all(promiseList).then(function (response) {
```

Comment:
Resolving them all together doesn't mean they've all been fulfilled, you're providing call back functions which are only called back when the request returns and you will never know it gets called back.